### PR TITLE
Core/Spells: PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW applies only with invisibility type INVISIBILITY_GENERAL (0)

### LINKED ISSUE #26688: Core/Spells: Invisibility auras and PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW
### Description

PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW is currently applied to any invisibility aura.

When comparing spells between retail and 3.3.5 that apply invisibility auras, I noticed that the spells (which already existed in 3.3.5) have an additional effect in retail:

> Effect X: Id 6 (SPELL_EFFECT_APPLY_AURA)
> Difficulty: Id 0 (DIFFICULTY_NONE)
> BasePoints = 0.00
> Targets (0, 0) (NO_TARGET, NO_TARGET)
> Aura Id 260 (SPELL_AURA_SCREEN_EFFECT), value = 0, misc = 81 (81), miscB = 1, amplitude = 0, periodic = 0
> ScreenEffect: Invis Screen Effect

This new effect I have only observed in spells that apply invisibility of the type INVISIBILITY_GENERAL (0). The screen effect "Invis Screen Effect" affects exactly the same way as the application of PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW.

Examples:
- https://www.wowhead.com/spell=3680/lesser-invisibility
- https://www.wowhead.com/spell=4986/unseen
- https://www.wowhead.com/spell=41253/greater-invisibility

### Expected behaviour

- [Master] Check if the new effect has replaced the application of PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW.
- [3.3.5] Maybe it could be assumed that PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW should only be applied with invisibility auras of type INVISIBILITY_GENERAL (0).
```diff
diff --git a/src/server/game/Spells/Auras/SpellAuraEffects.cpp b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
index 3dc138b379..16eb497177 100644
--- a/src/server/game/Spells/Auras/SpellAuraEffects.cpp
+++ b/src/server/game/Spells/Auras/SpellAuraEffects.cpp
@@ -1361,7 +1361,7 @@ void AuraEffect::HandleModInvisibility(AuraApplication const* aurApp, uint8 mode
     if (apply)
     {
         // apply glow vision
-        if (target->GetTypeId() == TYPEID_PLAYER)
+        if (target->GetTypeId() == TYPEID_PLAYER && type == INVISIBILITY_GENERAL)
             target->SetByteFlag(PLAYER_FIELD_BYTES2, PLAYER_FIELD_BYTES_2_OFFSET_AURA_VISION, PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW);
 
         target->m_invisibility.AddFlag(type);
@@ -1372,7 +1372,7 @@ void AuraEffect::HandleModInvisibility(AuraApplication const* aurApp, uint8 mode
         if (!target->HasAuraType(SPELL_AURA_MOD_INVISIBILITY))
         {
             // if not have different invisibility auras.
-            // remove glow vision
+            // always remove glow vision
             if (target->GetTypeId() == TYPEID_PLAYER)
                 target->RemoveByteFlag(PLAYER_FIELD_BYTES2, PLAYER_FIELD_BYTES_2_OFFSET_AURA_VISION, PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW);
 
@@ -1391,7 +1391,14 @@ void AuraEffect::HandleModInvisibility(AuraApplication const* aurApp, uint8 mode
                 }
             }
             if (!found)
+            {
+                // if not have invisibility auras of type INVISIBILITY_GENERAL
+                // remove glow vision
+                if (target->GetTypeId() == TYPEID_PLAYER && type == INVISIBILITY_GENERAL)
+                    target->RemoveByteFlag(PLAYER_FIELD_BYTES2, 3, PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW);
+
                 target->m_invisibility.DelFlag(type);
+            }
         }
 
         target->m_invisibility.AddValue(type, -GetAmount());
```

### Steps to reproduce the problem

1. Apply any invisibility aura on the player.

### Branch

3.3.5

### TC rev. hash/commit

https://github.com/TrinityCore/TrinityCore/commit/67bd11f5ffe17cb2daeb5074d92ca410eb269b09

### Operating system

Windows 10 x64

### Custom changes

None

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW should only be applied with invisibility auras of type INVISIBILITY_GENERAL (0).

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

**Issues addressed:**

Closes #26688

**Tests performed:**
Builds, tested in-game

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] [Master] Check if the new effect has replaced the application of PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW (more info here #26688).

### CODE REVIEW

File: src/server/game/Spells/Auras/SpellAuraEffects.cpp [hunk 1]
```diff
@@ -1391,7 +1391,14 @@ void AuraEffect::HandleModInvisibility(AuraApplication const* aurApp, uint8 mode
                 }
             }
             if (!found)
+            {
+                // if not have invisibility auras of type INVISIBILITY_GENERAL
+                // remove glow vision
+                if (target->GetTypeId() == TYPEID_PLAYER && type == INVISIBILITY_GENERAL)
+                    target->RemoveByteFlag(PLAYER_FIELD_BYTES2, 3, PLAYER_FIELD_BYTE2_INVISIBILITY_GLOW);
```
Comment (jackpoz) [inline src/server/game/Spells/Auras/SpellAuraEffects.cpp:1398]: Is there an enum value instead of "3" ?

Comment (meji46) [reply src/server/game/Spells/Auras/SpellAuraEffects.cpp]: Fixed!

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Comment by kelno** [comment]
One case where this is wrong: Spell 44856 uses INVISIBILITY_UNK10 and should also have the glow applied.

**Pull request merged**